public/API/API_GetAchievementsEarnedBetween.php

```diff
@@ -25,7 +25,13 @@
  *    string     GameURL                  site-relative path to the game page
  */
 
-$user = request()->query('u');
+use App\Site\Models\User;
+
+$user = User::firstWhere('User', request()->query('u'));
+if (!$user) {
+    return response()->json([]);
+}
+
```

Comment:
We could introduce validation responses here similar to the public request scripts. Was planned to happen in future iterations of the API:
```php
$input = Validator::validate(Arr::wrap(request()->post()), [
    'u' => 'required|string|exists:UserAccounts,User|alpha_num|max:20',
]);
```
i'm not sure what the framework's json response would look like and if we'd want to keep it this way.

Reply:
It returns
```
{
      "message": "The u field is required.",
      "errors": {
          "u": [
              "The u field is required."
          ]
      }
  }
```

Instead of 
```
[]
```

So, instead of an empty array, the client would receive an error object.

Reply:
Additionally, doesn't the `exists` validation have the query the database, thus adding additional overhead since we have to retrieve the record anyway?

Reply:
good point, exists is really fast though.
let's revisit the validation thing sepatately when refactoring the v2 web api into controllers.

apart from that, is empty array the correct response, should it be empty object?

Reply:
nevermind, in this case it's an array 👍